pubsub/pulsar/pulsar.go

```diff
@@ -217,6 +226,14 @@ func (p *Pulsar) Publish(ctx context.Context, req *pubsub.PublishRequest) error
 			opts.Schema = getPulsarSchema(sm)
 		}
 
+		if p.useProducerEncryption() {
+			keyReader := crypto.NewFileKeyReader(p.metadata.PublicKey, "")
```

Comment:
Please also provide a way for the keys themselves to be passed as value for the metadata properties, not just a path to a file to disk (eg the value of the property being the actual PEM-encoded string). We do the same for other components such as the HTTP binding.

Reply:
This isn't possible as the Pulsar SDK only takes file paths.

Reply:
I agree with that concern - mounting files into containers in various container environments is really a hassle that is best avoided.